src/Lib/Managers/SelectedBoundManager.vala

```diff
@@ -421,4 +415,76 @@ public class Akira.Lib.Managers.SelectedBoundManager : Object {
             snap_manager.populate_decorators_from_data (matches, snap_grid);
         }
     }
+
+    private void scale_from_event (
+        Lib.Items.CanvasItem item,
+        Managers.NobManager.Nob selected_nob,
+        double event_x,
+        double event_y
+    ) {
+        if (!initial_drag_registered) {
+            initial_drag_registered = true;
+            initial_drag_item_x = item.transform.x;
+            initial_drag_item_y = item.transform.y;
+            item.get_transform (out initial_item_transform);
+        }
+
+        double rel_event_x = event_x;
+        double rel_event_y = event_y;
+        double rel_press_x = initial_drag_press_x;
+        double rel_press_y = initial_drag_press_y;
+
+        var canvas = (Lib.Canvas) item.canvas;
+        // Convert the coordinates from the canvas to the item so we know the real
+        // values even if the item is rotated.
+        canvas.convert_to_item_space (item, ref rel_event_x, ref rel_event_y);
+        canvas.convert_to_item_space (item, ref rel_press_x, ref rel_press_y);
+
+        // Calculate the change based on the event.
+        var delta_x = rel_event_x - rel_press_x;
+        var delta_y = rel_event_y - rel_press_y;
+
+        double item_x = item.bounds.x1;
+        double item_y = item.bounds.y1;
+        canvas.convert_to_item_space (item, ref item_x, ref item_y);
+
+        bool ratio_locked = canvas.ctrl_is_pressed || item.size.locked;
+
+        // these values will be populated
+        double inc_width = 0;
+        double inc_height = 0;
+        double inc_x = 0;
+        double inc_y = 0;
+        bool h_flip = false;
+        bool v_flip = false;
+
+        Utils.AffineTransform.calculate_size_adjustments (
+            selected_nob,
+            initial_width,
+            initial_height,
+            delta_x,
+            delta_y,
+            initial_width / initial_height,
```

Comment:
The size ratio can be accessed from the `Size` component without needing of calculation.
`item.size.ratio`

Reply:
I preferred using the initial values for accuracy, otherwise resize weirdness could cause issues? I am not sure if it will, but this feels a bit more robust. But I can change it if you think it's better.

Reply:
That's okay with me.